google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcParameterStoreTest.java

```diff
@@ -0,0 +1,461 @@
+    verifyParameter(params, Value.boolArray(new boolean[] {true, false, true}));
+
+    params.setParameter(1, JdbcArray.createArray("INT64", new Long[] {1L, 2L, 3L}), Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("INT64", new Long[] {1L, 2L, 3L}))));
+    verifyParameter(params, Value.int64Array(new long[] {1, 2, 3}));
+
+    params.setParameter(1, JdbcArray.createArray("FLOAT64", new Double[] {1D, 2D, 3D}),
+        Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("FLOAT64", new Double[] {1D, 2D, 3D}))));
+    verifyParameter(params, Value.float64Array(new double[] {1, 2, 3}));
+
+    @SuppressWarnings("deprecation")
+    Date sqlDate = new Date(2018 - 1900, 12 - 1, 14);
+    params.setParameter(1, JdbcArray.createArray("DATE", new Date[] {sqlDate}), Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("DATE", new Date[] {sqlDate}))));
+    verifyParameter(params,
+        Value.dateArray(Arrays.asList(com.google.cloud.Date.fromYearMonthDay(2018, 12, 14))));
+
+    Timestamp sqlTimestamp = new Timestamp(System.currentTimeMillis());
+    params.setParameter(1, JdbcArray.createArray("TIMESTAMP", new Timestamp[] {sqlTimestamp}),
+        Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("TIMESTAMP", new Timestamp[] {sqlTimestamp}))));
+    verifyParameter(params,
+        Value.timestampArray(Arrays.asList(com.google.cloud.Timestamp.of(sqlTimestamp))));
+
+    params.setParameter(1, JdbcArray.createArray("BYTES", new byte[][] {{1, 2, 3}, {4, 5, 6}}),
+        Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("BYTES", new byte[][] {{1, 2, 3}, {4, 5, 6}}))));
+    verifyParameter(params, Value.bytesArray(Arrays.asList(ByteArray.copyFrom(new byte[] {1, 2, 3}),
+        ByteArray.copyFrom(new byte[] {4, 5, 6}))));
+
+    params.setParameter(1,
+        JdbcArray.createArray("STRING", new String[] {"test1", "test2", "test3"}), Types.ARRAY);
+    assertThat((JdbcArray) params.getParameter(1),
+        is(equalTo(JdbcArray.createArray("STRING", new String[] {"test1", "test2", "test3"}))));
+    verifyParameter(params, Value.stringArray(Arrays.asList("test1", "test2", "test3")));
+  }
+
+  private void verifyParameter(JdbcParameterStore params, Value value) throws SQLException {
+    Statement.Builder builder = Statement.newBuilder("SELECT * FROM FOO WHERE BAR=:p1");
+    params.bindParameterValue(builder.bind("p1"), 1);
+    assertThat(builder.build().getParameters().get("p1"), is(equalTo(value)));
+  }
+
+  @Test
+  public void testConvertPositionalParametersToNamedParameters() throws SQLException {
+    assertThat(
+        convertPositionalParametersToNamedParameters(
+            "select * from foo where name=?").sqlWithNamedParameters,
```

Comment:
can we add some more basic tests like this statement?

Reply:
Added additional test statements. Is there anything specific that you had in mind?

Reply:
That looks good, thanks!